src/main/java/org/example/inspector/SongInspectorImpl.java

```diff
@@ -0,0 +1,92 @@
+package org.example.inspector;
+
+import com.google.common.collect.ImmutableSet;
+import ealvatag.audio.AudioFile;
+import ealvatag.audio.AudioFileIO;
+import ealvatag.audio.exceptions.CannotReadException;
+import ealvatag.audio.exceptions.InvalidAudioFrameException;
+import ealvatag.tag.FieldKey;
+import ealvatag.tag.NullTag;
+import ealvatag.tag.Tag;
+import ealvatag.tag.TagException;
+import org.apache.commons.io.FilenameUtils;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+
+public class SongInspectorImpl implements SongInspector {
+  private final String FLAC = "flac";
+  private final String MP3 = "mp3";
+
+  public static File getCurrentDirectory() {
+    final Path path = FileSystems.getDefault().getPath("").toAbsolutePath();
+    return new File(String.valueOf(path));
+  }
+
+  private boolean isMediaFile(final File file) {
+    if (file.isFile()) {
+      final String extension = FilenameUtils.getExtension(file.getName()).toLowerCase();
```

Comment:
Adding a whole library just to get an extension is no good. It's really 2-5 lines of code.